### PR TITLE
WIP fs: Make it so mark_written has to be used

### PR DESCRIPTION
**Pull request by AliceGrey**
Changed the semantics of fs_creat/fs_creat_replacing to always wait to clear tmp_file until fs_mark_written.

### CODE REVIEW

File: rcore/fs_test.c [hunk 1]
```diff
@@ -74,5 +76,54 @@ TEST(fs_two_files) {
...
+TEST(fs_replace_file_basic) {
+    struct fd fd, *fdp;
+    struct file file;
+    int rv = TEST_PASS;
+
+    *artifact = 0;
+
+    //create a file with contents a
+    fdp = fs_creat(&fd, "hello", 5);
+       if (!fdp) { *artifact = 1; return TEST_FAIL; }
```
Comment (jwise) [inline rcore/fs_test.c]: The indentation here is a little bizarre.

File: rcore/fs_test.c [hunk 2]
```diff
@@ -74,5 +76,54 @@ TEST(fs_two_files) {
...
+       fs_write(&fd, "Hello", 5);
```
Comment (jwise) [inline rcore/fs_test.c]: Needs `fs_mark_written`?

**Comment by AliceGrey** [comment]
Oh yeah res pack must have been pushed by accident. I think I rebased this branch on another non-master branch.

**Comment by jwise** [comment]
Rebased and cleaned up a little bit.  Thanks!

**Pull request merged**